rust/lance-io/src/object_store.rs

```diff
@@ -396,6 +406,7 @@ impl ObjectStore {
                 inner: Arc::new(LocalFileSystem::new()).traced(),
                 scheme: String::from(scheme),
                 block_size: 4 * 1024, // 4KB block size
+                io_parallelism: DEFAULT_LOCAL_IO_PARALLELISM,
```

Comment:
Note: I think I might move this into `LimitStore`, so we can use the centralized semaphores there.

Reply:
Hmm, that might be tricky.  The scan scheduler uses a priority queue when I/O parallelism sempahore is full.  I'm not sure if limit store would enable us to do the same thing.

Reply:
Oh wait, I was thinking of `ScanScheduler`.  Moving this in `LimitStore` is fine with me.